From 2b6b4520e5c23254fdc4936d49b08d3995be957e Mon Sep 17 00:00:00 2001
From: Pekka Lundstrom <pekka.lundstrom@jollamobile.com>
Date: Tue, 16 Apr 2013 14:16:15 +0300
Subject: [PATCH] Ensure machine-id in start

Signed-off-by: Pekka Lundstrom <pekka.lundstrom@jollamobile.com>
---
 bus/dbus.service.in |    1 +
 1 file changed, 1 insertion(+)

diff --git a/bus/dbus.service.in b/bus/dbus.service.in
index 160947c..4d56b30 100644
--- a/bus/dbus.service.in
+++ b/bus/dbus.service.in
@@ -4,6 +4,7 @@ Requires=dbus.socket
 After=syslog.target
 
 [Service]
+ExecStartPre=@EXPANDED_BINDIR@/dbus-uuidgen --ensure
 ExecStart=@EXPANDED_BINDIR@/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation
 ExecReload=@EXPANDED_BINDIR@/dbus-send --print-reply --system --type=method_call --dest=org.freedesktop.DBus / org.freedesktop.DBus.ReloadConfig
 OOMScoreAdjust=-900
-- 
1.7.9.5

