From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Carsten Munk <carsten.munk@jollamobile.com>
Date: Thu, 17 Oct 2013 07:52:09 +0000
Subject: [PATCH] Disable setuid checking due to it conflicting with setgid

Signed-off-by: Carsten Munk <carsten.munk@jollamobile.com>
---
 dbus/dbus-sysdeps-unix.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dbus/dbus-sysdeps-unix.c b/dbus/dbus-sysdeps-unix.c
index 8743589f..a3ce283c 100644
--- a/dbus/dbus-sysdeps-unix.c
+++ b/dbus/dbus-sysdeps-unix.c
@@ -4821,6 +4821,7 @@ _dbus_fd_set_all_close_on_exec (void)
 dbus_bool_t
 _dbus_check_setuid (void)
 {
+  return FALSE;
   /* TODO: get __libc_enable_secure exported from glibc.
    * See http://www.openwall.com/lists/owl-dev/2012/08/14/1
    */
