From 6dad652fb785e99adab8898b6348ab268cab5b68 Mon Sep 17 00:00:00 2001
From: Niels Breet <niels.breet@jolla.com>
Date: Mon, 29 Jul 2019 14:10:55 +0300
Subject: [PATCH] Specify configdir correctly

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 21c50d41..47aeb147 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -464,7 +464,7 @@ else(WIN32)
     set(DBUS_SESSION_BUS_LISTEN_ADDRESS "unix:tmpdir=${DBUS_SESSION_SOCKET_DIR}" CACHE STRING "session bus default listening address")
     set(DBUS_SESSION_BUS_CONNECT_ADDRESS "autolaunch:" CACHE STRING "session bus fallback address for clients")
     set(sysconfdir "")
-    set(configdir ${sysconfdir}/dbus-1 )
+    set(configdir /usr/share/dbus-1)
     set(DBUS_SYSTEM_CONFIG_FILE  ${configdir}/system.conf)
     set(DBUS_SESSION_CONFIG_FILE ${configdir}/session.conf)
     set(DBUS_USER "messagebus")
-- 
2.20.1

