From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matti Kosola <matti.kosola@jolla.com>
Date: Fri, 16 Aug 2019 09:15:06 +0300
Subject: [PATCH] Enable building with libaudit

Signed-off-by: Matti Kosola <matti.kosola@jolla.com>
---
 CMakeLists.txt      | 6 ++++++
 dbus/CMakeLists.txt | 3 +++
 2 files changed, 9 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 61f4715d949ba38cd2f82adcd44ad08897abbb3f..abdfa78e705447d6510f7bfaa9e7df719cc455a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -636,6 +636,11 @@ set(DBUS_DAEMON_NAME "dbus-daemon" CACHE STRING "The name of the dbus daemon exe
 set(HAVE_SELINUX 1)
 add_definitions(-DHAVE_SELINUX)
 
+# Enable libaudit support unconditionally. --mkosola
+
+set(HAVE_LIBAUDIT 1)
+add_definitions(-DHAVE_LIBAUDIT)
+
 ########### create config.h ###############
 
 #include(ConfigureChecks.cmake)
@@ -780,6 +785,7 @@ message("        Session bus listens on:   ${DBUS_SESSION_BUS_LISTEN_ADDRESS}  "
 message("        Session clients connect to: ${DBUS_SESSION_BUS_CONNECT_ADDRESS} ")
 else()
 message("        SELinux support:          ${HAVE_SELINUX}                     ")
+message("        Libaudit support:         ${HAVE_LIBAUDIT}                    ")
 message("        System bus address:       ${DBUS_SYSTEM_BUS_DEFAULT_ADDRESS}  ")
 message("        System bus socket:        ${DBUS_SYSTEM_SOCKET}               ")
 message("        System bus PID file:      ${DBUS_SYSTEM_PID_FILE}             ")
diff --git a/dbus/CMakeLists.txt b/dbus/CMakeLists.txt
index d0656349c3008e51e9bedeefa85d9340f538522e..e82acc11e824ff709349a460f89c8dee614ee0da 100644
--- a/dbus/CMakeLists.txt
+++ b/dbus/CMakeLists.txt
@@ -300,6 +300,9 @@ else(WIN32)
     if(HAVE_SELINUX)
         target_link_libraries(dbus-1 selinux)
     endif()
+    if(HAVE_LIBAUDIT)
+        target_link_libraries(dbus-1 audit cap-ng)
+    endif()
 endif()
 
 # target definitions passed to the clients
