From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Juho=20H=C3=A4m=C3=A4l=C3=A4inen?=
 <juho.hamalainen@jolla.com>
Date: Wed, 5 Feb 2020 11:08:07 +0200
Subject: [PATCH] Check for monotonic clock.

---
 CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7cf9c01a..7d724075 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -597,6 +597,14 @@ if(UNIX)
     endif()
 endif()
 
+if(UNIX AND CMAKE_HAVE_PTHREAD_H)
+    include(CheckSymbolExists)
+    check_symbol_exists(CLOCK_MONOTONIC time.h HAVE_CLOCK_MONOTONIC)
+    if(HAVE_CLOCK_MONOTONIC)
+        add_definitions(-DHAVE_MONOTONIC_CLOCK)
+    endif()
+endif()
+
  # Not used on Windows, where there is no system bus
 set(DBUS_SYSTEM_PID_FILE ${DBUS_RUNSTATEDIR}/dbus/pid)
 
